receiver/elasticapmreceiver/internal/mappers/intakeV2ToDerivedFields.go

```diff
@@ -89,3 +90,69 @@ func SetDerivedFieldsCommon(event *modelpb.APMEvent, attributes pcommon.Map) {
 		attributes.PutStr(elasticattr.EventOutcome, "unknown")
 	}
 }
+
+// Sets fields that are NOT part of OTel for errors. These fields are derived by the Enrichment lib in case of OTLP input
+func SetDerivedFieldsForError(event *modelpb.APMEvent, attributes pcommon.Map) {
+	attributes.PutStr(elasticattr.ProcessorEvent, "error")
+
+	if event.Error == nil {
+		return
+	}
+
+	if event.Error.Id != "" {
+		attributes.PutStr(elasticattr.ErrorID, event.Error.Id)
+	}
+	if event.Transaction != nil && event.Transaction.Id != "" {
+		attributes.PutStr(elasticattr.TransactionID, event.Transaction.Id)
+	}
+	if event.ParentId != "" {
+		attributes.PutStr(elasticattr.ParentID, event.ParentId)
+	}
+
+	if event.Error.Type != "" {
+		attributes.PutStr("error.type", event.Error.Type)
+	}
+	if event.Error.Message != "" {
+		attributes.PutStr("message", event.Error.Message)
+	}
+	attributes.PutStr(elasticattr.ErrorGroupingKey, event.Error.GroupingKey)
+	attributes.PutInt(elasticattr.TimestampUs, int64(event.Timestamp/1_000))
+
+	if event.Error.Culprit != "" {
+		attributes.PutStr("error.culprit", event.Error.Culprit)
+	}
+
+	if event.Error.Exception != nil {
+		if event.Error.Exception.Type != "" {
+			attributes.PutStr("exception.type", event.Error.Exception.Type)
+		}
+		if event.Error.Exception.Message != "" {
+			attributes.PutStr("exception.message", event.Error.Exception.Message)
+		}
+
+		if event.Error.Exception.Stacktrace != nil {
+			str := ""
+			for _, frame := range event.Error.Exception.Stacktrace {
+				if frame.Function != "" {
+					str += fmt.Sprintf("%s:%d %s \n", frame.Filename, *frame.Lineno, frame.Function)
+				} else {
+					if frame.Classname != "" && frame.Lineno != nil {
+						str += fmt.Sprintf("%s:%d \n", frame.Classname, *frame.Lineno)
```

Comment:
Are the trailing spaces needed?

Reply:
Yeah, this has a history :) 

Originally I did not have those, but then golden formatted the generated yaml files in a way that never matched the actual data in `plogtest.CompareLogs`.  It was super strange. So that's why I ended up with this.

Reply:
Great, thanks for the context!

A follow-up suggestion would be to use the `protobufs` auto generated getters for non-checked pointer values (if they could be nil), something like:

```go
				if frame.Function != "" {
					str += fmt.Sprintf("%s:%d %s \n", frame.GetFilename(), frame.GetLineno(), frame.Function)
				} else {
					if frame.Classname != "" && frame.Lineno != nil {
						str += fmt.Sprintf("%s:%d \n", frame.Classname, *frame.Lineno)
					}
				}
```